Source/monster.cpp

```diff
@@ -2085,22 +2080,18 @@ void AiRanged(int monsterId)
 	}
 
 	if (monster.activeForTicks == UINT8_MAX || (monster.flags & MFLAG_TARGETS_MONSTER) != 0) {
-		int fx = monster.enemyPosition.x;
-		int fy = monster.enemyPosition.y;
-		int mx = monster.position.tile.x - fx;
-		int my = monster.position.tile.y - fy;
 		Direction md = GetMonsterDirection(monster);
 		if (monster.activeForTicks < UINT8_MAX)
 			MonstCheckDoors(monster);
 		monster.direction = md;
 		if (static_cast<MonsterMode>(monster.var1) == MonsterMode::RangedAttack) {
 			AiDelay(monster, GenerateRnd(20));
-		} else if (abs(mx) < 4 && abs(my) < 4) {
+		} else if (monster.longestDistanceToEnemy() < 4) {
 			if (GenerateRnd(100) < 10 * (monster.intelligence + 7))
 				RandomWalk(monsterId, Opposite(md));
 		}
 		if (monster.mode == MonsterMode::Stand) {
-			if (LineClearMissile(monster.position.tile, { fx, fy })) {
+			if (LineClearMissile(monster.position.tile, { monster.enemyPosition.x, monster.enemyPosition.y })) {

```

Comment:
It's already a Point :)
```suggestion
			if (LineClearMissile(monster.position.tile, monster.enemyPosition)) {
```

Reply:
Ah! nice!